### PR TITLE
Add `declare_cairo0` libfunc test

### PR DESCRIPTION
**Pull request by Arcticae**
Should be merged after https://github.com/software-mansion-labs/cairo/pull/38

### CODE REVIEW

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 1]
```diff
@@ -23,6 +23,11 @@ def get_mock_for_lib_func(
         return_value = type(
             "return_value", (object,), {"err_code": err_code, "ok": ok}
         )()
+    if lib_func_name == "declare_legacy":
+        ok = type("ok", (object,), {"class_hash": 0})()
+        return_value = type(
+            "return_value", (object,), {"err_code": err_code, "ok": ok}
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py:29]: I don't know what you mean, we don't do such thing for declare

You mean output validation? Input validation?

[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py:29]: I meant invoke, and for some reason my brain was sure declare takes calldata, nevermind, sorry

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 2]
```diff
@@ -23,6 +23,11 @@ def get_mock_for_lib_func(
         return_value = type(
             "return_value", (object,), {"err_code": err_code, "ok": ok}
         )()
+    elif lib_func_name == "declare_legacy":
```
Comment (karol-bisztyga) [inline tests/integration/cairo1_compiler_bindings/library_functions_test.py:26]: why don't we do:
```
if lib_func_name == "declare" or lib_func_name == "declare_legacy":
```
those ret values are the same, right?

**Review by karol-bisztyga — COMMENTED** [review]
Wouldn't it be better to name it `declare_cairo0` instead of "legacy (sorry I didn't bring this up in https://github.com/software-mansion-labs/cairo/pull/36, I just had this thought now and I don't feel strongly about this)?

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py (hunk 2 above)
Comment (Arcticae) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py]: [17b279e](https://github.com/software-mansion/protostar/pull/1622/commits/17b279e39d758f236aae615516011c107bf59d10)

**Review by karol-bisztyga — APPROVED** [review]

**Comment by Arcticae** [comment]
Will fail until #1632 is merged

**Pull request merged**